modules/images/webp-uploads/load.php

```diff
@@ -11,42 +11,106 @@
+ * top level of the image metadata to store the references for all the mime types for the `full` size image of the
+ * attachment.
  *
  * @since n.e.x.t
  *
  * @see   wp_generate_attachment_metadata()
- * @see   webp_uploads_valid_image_mime_types()
+ * @see   webp_uploads_get_supported_image_mime_transforms()
  *
  * @param array $metadata      An array with the metadata from this attachment.
  * @param int   $attachment_id The ID of the attachment where the hook was dispatched.
  * @return array An array with the updated structure for the metadata before is stored in the database.
  */
 function webp_uploads_create_sources_property( array $metadata, $attachment_id ) {
-	// Make sure we have some sizes to work with, otherwise avoid any work.
-	if ( empty( $metadata['sizes'] ) || ! is_array( $metadata['sizes'] ) ) {
-		return $metadata;
-	}
 	// This should take place only on the JPEG image.
 	$valid_mime_transforms = webp_uploads_get_supported_image_mime_transforms();
-
 	// Not a supported mime type to create the sources property.
 	$mime_type = get_post_mime_type( $attachment_id );
-	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) ) {
+	if ( ! isset( $valid_mime_transforms[ $mime_type ] ) || ! is_array( $valid_mime_transforms[ $mime_type ] ) ) {
 		return $metadata;
 	}
 
 	$file = get_attached_file( $attachment_id, true );
-
 	// File does not exist.
 	if ( ! file_exists( $file ) ) {
 		return $metadata;
 	}
+	// Make sure the top level `sources` key is a valid array.
+	if ( ! isset( $metadata['sources'] ) || ! is_array( $metadata['sources'] ) ) {
+		$metadata['sources'] = array();
+	}
+
+	if ( empty( $metadata['sources'][ $mime_type ] ) ) {
+		$metadata['sources'][ $mime_type ] = array(
+			'file'     => wp_basename( $file ),
+			'filesize' => filesize( $file ),
+		);
+		wp_update_attachment_metadata( $attachment_id, $metadata );
+	}
+
+	$allowed_mimes      = array_flip( wp_get_mime_types() );
+	$original_directory = pathinfo( $file, PATHINFO_DIRNAME );
+	$original_extension = pathinfo( $file, PATHINFO_EXTENSION );
+	// Create the sources for the full sized image.
+	foreach ( $valid_mime_transforms[ $mime_type ] as $targeted_mime ) {
+		// If this property exists no need to create the image again.
+		if ( ! empty( $metadata['sources'][ $targeted_mime ] ) ) {
+			continue;
+		}
+
+		if ( ! isset( $allowed_mimes[ $targeted_mime ] ) || ! is_string( $allowed_mimes[ $targeted_mime ] ) ) {
+			continue;
+		}
+
+		if ( ! wp_image_editor_supports( array( 'mime_type' => $targeted_mime ) ) ) {
+			continue;
+		}
+
+		$editor = wp_get_image_editor( $file );
+
+		if ( is_wp_error( $editor ) ) {
+			continue;
+		}
+
+		if ( ! isset( $metadata['width'], $metadata['height'] ) ) {
+			continue;
+		}
+
+		$editor->resize( $metadata['width'], $metadata['height'], false );
+		$extension = explode( '|', $allowed_mimes[ $targeted_mime ] );
+		$extension = reset( $extension );
```

Comment:
It would be more straightforward to do the following:
```suggestion
		$extension = $extension[0];
```